doc/build/snippets/design.rst

```diff
@@ -0,0 +1,60 @@
+Snippets Design
+###############
+
+This page documents design considerations related to snippets.
+
+List of Design Goals
+********************
+
+- **extensible**: for example, it is possible to add board support for an
+  existing built-in snippet without modifying the zephyr repository
+
+- **composable**: it is possible to use multiple snippets at once, for example
+  using:
+
+  .. code-block:: console
+
+     west build -S <snippet1> -S <snippet2> ...
+
+- **able to combine multiple types of configuration**: snippets make it possible
+  to store multiple different types of build system settings in one place, and
+  apply them all together
+
+- **specializable**: for example, it is possible to customize a snippet's
+  behavior for a particular board, or board revision
+
+- **future-proof and backwards-compatible**: arbitrary future changes to the
+  snippets feature will be possible without breaking backwards compatibility
+  for older snippets
+
+- **applicable to purely "software" changes**: unlike the shields feature,
+  snippets do not assume the presence of a "daughterboard", "shield", "hat", or
+  any other type of external assembly which is connected to the main board
+
+- **DRY** (don't repeat yourself): snippets allow you to skip unnecessary
+  repetition; for example, you can apply the same board-specific configuration
+  to boards ``foo`` and ``bar`` by specifying ``/(foo|bar)/`` as a regular
+  expression for the settings, which wil then apply to both boards
+
+Comparison with Shields
+***********************
+
+Snippets were originally developed to bring the :ref:`shields` feature set to a
+wider context, and to extend some of the shields feature's design limitations.
+
+As a result, snippets will eventually become a strict generalization of
+shields: when this work is complete, it will be possible to specify any shield
+as a snippet. The reverse is not be true: not all snippets can be "expressed"
+as a shield.
+
+Here is a list of some limitations of the shields feature not present in the
+snippets feature:
+
+- **limited to external ICs**: at least in concept, shields require
+  some sort of external hardware
+- **not extensible**: adding board support for a shield defined in
+  :zephyr_file:`zephyr/boards/shields/` requires modifying the zephyr
+  repository itself
+- **not DRY**: for example, shields require identical board-specific
+  devicetree overlays to be specified via multiple identical files on
+  the file system
```

Comment:
To unblock this PR, please remove "Comparison with Shields" chapter. It is incorrect and gives a false sense of the state of the shields and current snippet support.